qml/Launcher/Drawer.qml

```diff
@@ -29,12 +29,85 @@ FocusScope {
     readonly property bool moving: appList && appList.moving
     readonly property Item searchTextField: searchField
     readonly property real delegateWidth: units.gu(10)
+    visible: x > -width
+    property var fullyOpen: x === 0
+    property var fullyClosed: x === -width
 
     signal applicationSelected(string appId)
 
+    // Request that the Drawer is opened fully, if it was partially closed then
+    // brought back
+    signal open();
+
+    // Request that the Drawer (and maybe its parent) is hidden, normally if
+    // the Drawer has been dragged away.
+    signal hide();
```

Comment:
I would call these `openRequested` and `hideRequested`, so that on the receiving side you'll have `onOpenRequested` and `onHideRequested`. `open()` and `hide()` are good names for functions, not for signals (see also `applicationSelected` above: it's not just `select`).

Reply:
Also, remove the `;` at the end of those lines